tests/MongoDB.Driver.Core.Tests/Core/Operations/OperationTestBase.cs

```diff
@@ -398,61 +400,111 @@ protected void Update(string filter, string update)
                 using (var binding = new WritableServerBinding(cluster, session.Fork()))
                 {
                     var cancellationToken = new CancellationTokenSource().Token;
+                    Exception exception;
                     if (async)
                     {
-                        executeAsync(binding, cancellationToken).GetAwaiter().GetResult();
+                        exception = Record.Exception(() => executeAsync(binding, cancellationToken).GetAwaiter().GetResult());
                     }
                     else
                     {
-                        execute(binding, cancellationToken);
+                        exception = Record.Exception(() => execute(binding, cancellationToken));
                     }
 
-                    var commandStartedEvent = (CommandStartedEvent)eventCapturer.Next();
-                    var command = commandStartedEvent.Command;
-                    if (session.Id == null)
-                    {
-                        command.Contains("lsid").Should().BeFalse();
-                    }
-                    else
-                    {
-                        command["lsid"].Should().Be(session.Id);
-                    }
-                    session.ReferenceCount().Should().Be(2);
+                    assertResults(eventCapturer, session, exception);
                 }
             }
         }
 
+        // private methods
+        private void AssertSessionIdIsSentIfPresented(EventCapturer eventCapturer, ICoreSessionHandle session, Exception exception)
+        {
+            exception.Should().BeNull();
+            var commandStartedEvent = (CommandStartedEvent)eventCapturer.Next();
+            var command = commandStartedEvent.Command;
+            if (session.Id == null)
+            {
+                command.Contains("lsid").Should().BeFalse();
+            }
+            else
+            {
+                command["lsid"].Should().Be(session.Id);
+            }
+
+            session.ReferenceCount().Should().Be(2);
+        }
+
+        private void AssertSessionIdWithUnacknowledgedWrite(EventCapturer eventCapturer, ICoreSessionHandle session, Exception ex)
+        {
+            if (session.IsImplicit)
+            {
+                var commandStartedEvent = (CommandStartedEvent)eventCapturer.Next();
+                var command = commandStartedEvent.Command;
+                command.Contains("lsid").Should().BeFalse();
+                session.ReferenceCount().Should().Be(2);
+            }
+            else
+            {
+                ex.Should().BeAssignableTo<NotSupportedException>();
+            }
+        }
+
+        private ICoreSessionHandle CreateSession(bool useExplicitSessionIfSupported, ICluster cluster)
+        {
+            return useExplicitSessionIfSupported
+                ? CoreTestConfiguration.StartSession(cluster)
+                : cluster.StartSession(new CoreSessionOptions(isImplicit: true));
+        }
```

Comment:
Refactor to let `CoreTestConfiguration` do most of the work:

    private ICoreSessionHandle CreateSession(ICluster cluster, bool useImplicitSession)
    {
        var options = new CoreSessionOptions(isImplicit: useImplicitSession);
        return CoreTestConfiguration.StartSession(cluster, options);
    }